src/TesApi.Web/AzureProxy.cs

```diff
@@ -478,6 +491,32 @@ public async Task DeleteBatchPoolIfExistsAsync(string poolId, CancellationToken
             }
         }
 
+        /// <inheritdoc/>
+        public Task<CloudPool> GetBatchPoolAsync(string poolId, DetailLevel detailLevel = default, CancellationToken cancellationToken = default)
+            => batchClient.PoolOperations.GetPoolAsync(poolId, detailLevel: detailLevel, cancellationToken: cancellationToken);
+
+        /// <inheritdoc/>
+        public Task CommitBatchPoolChangesAsync(CloudPool pool, CancellationToken cancellationToken = default)
+            => pool.CommitChangesAsync(cancellationToken: cancellationToken);
+
+        /// <inheritdoc/>
+        public async Task<(int? lowPriorityNodes, int? dedicatedNodes)> GetCurrentComputeNodesAsync(string poolId, CancellationToken cancellationToken = default)
+        {
+            var pool = await batchClient.PoolOperations.GetPoolAsync(poolId, detailLevel: new ODATADetailLevel(selectClause: "currentLowPriorityNodes,currentDedicatedNodes"), cancellationToken: cancellationToken);
+            return (pool.CurrentLowPriorityComputeNodes, pool.CurrentDedicatedComputeNodes);
+        }
+
+        /// <inheritdoc/>
+        public async Task<(AllocationState? AllocationState, bool? AutoScaleEnabled, int? TargetLowPriority, int? TargetDedicated)> GetComputeNodeAllocationStateAsync(string poolId, CancellationToken cancellationToken = default)
```

Comment:
Why not create a type for this -e.g. NodeAllocationState instead of multiple return values.?

Reply:
Technically it's a ValueTuple, and this is the only place this one is created, we don't ever compare these, and it's less typing to return one of these than it is to create (even records). However, I do see how this negatively affects readability.

I'll probably deal with this in a future PR, although I may also address it before this PR is wrapped up.